### PR TITLE
Implement `RRule::Rule#is_finite?`

### PR DESCRIPTION
**Pull request by bahriddin**
I am using gem to create repeating calendar events on DB. For that I want to check how many events do I need to create so I am using sth like this:

```Ruby
rrule = RRule::Rule.new("RRULE:FREQ=DAILY;INTERVAL=1")
puts rrule.count
```
When rule doesn't have ending rule it is calculating `count` using [Enumerable.count](https://apidock.com/ruby/Enumerable/count) and it is quite slow to count events between now and max_date (year 9999).

For this reason I believe having an interface to check if it has end limit will be handy.

### CODE REVIEW

File: lib/rrule/rule.rb [hunk 1]
```diff
@@ -87,6 +87,10 @@ def humanize
       Humanizer.new(self, options).to_s
     end
 
+    def has_end_limit?
```
Comment (rmitchell-sq) [inline lib/rrule/rule.rb:90]: Nice, thanks! One nit - what about calling this method `is_finite?`? "limit" puts me in mind of a SQL numerical LIMIT and I think I might wonder if this method conveyed the same (i.e. if it meant "this RRULE has a COUNT").

**Review by rmitchell-sq — APPROVED** [review]

File: lib/rrule/rule.rb (hunk 1 above)
Comment (bahriddin) [reply lib/rrule/rule.rb:90]: Yeah, good point. Updating..

File: Gemfile [hunk 2]
```diff
@@ -2,13 +2,13 @@
...
 
-gem 'pry', '~> 0.12.2'
+gem 'pry', '>= 0.13'
 gem 'rake'
 gem 'rspec', '~> 3.8'
 gem 'rubocop', '0.63.1'
 
 platform :mri do
-  gem 'pry-byebug'
+  gem 'pry-byebug', "~> 3.10.1"
```
Comment (bahriddin) [inline Gemfile:11]: I had to update them as I was getting the following error when running specs:
> undefined method `=~' for class `Pry::Code'

Related: https://gitlab.com/gitlab-org/customers-gitlab-com/-/issues/6143

**Comment by rmitchell-sq** [comment]
LGTM, thank you!

**Pull request merged**